packages/frontend/components/profile/user-account/my-list/component.tsx

```diff
@@ -89,103 +89,44 @@ export const MyListInfo = () => {
-          <div css={styles.header}>
-            <SectionHeader>{t('my-list:card.purchased')}</SectionHeader>
-          </div>
-          <div css={styles.container}>
-            {typedItems({
-              items: filteredItems,
-              byStatus: ProductStatus.FINISHED,
-              byKey: 'author',
-            }).map((item) => (
-              <Purchased key={item.id} data={item} />
-            ))}
-          </div>
-        </div>
+      {filteredItems.purchased && filteredItems.purchased.length ? (
+        <ProductSection
+          headerText={t('my-list:card.purchased')}
+          items={filteredItems.purchased}
+          Card={Purchased}
+        />
       ) : null}
 
-      {typedItems({
-        items: filteredItems,
-        byStatus: ProductStatus.FINISHED,
-        byKey: 'winner',
-      }).length ? (
-        <div css={styles.section}>
-          <div css={styles.header}>
-            <SectionHeader>{t('my-list:card.sold')}</SectionHeader>
-          </div>
-          <div css={styles.container}>
-            {typedItems({
-              items: filteredItems,
-              byStatus: ProductStatus.FINISHED,
-              byKey: 'winner',
-            }).map((item) => (
-              <Sold key={item.id} data={item} />
-            ))}
-          </div>
-        </div>
+      {filteredItems.sold && filteredItems.sold.length ? (
+        <ProductSection
+          headerText={t('my-list:card.sold')}
+          items={filteredItems.sold}
+          Card={Sold}
+        />
       ) : null}
 
-      {typedItems({
-        items: filteredItems,
-        byStatus: ProductStatus.ACTIVE,
-      }).length ? (
-        <div css={styles.section}>
-          <div css={styles.header}>
-            <SectionHeader>{t('my-list:card.posted')}</SectionHeader>
-          </div>
-          <div css={styles.container}>
-            {typedItems({
-              items: filteredItems,
-              byStatus: ProductStatus.ACTIVE,
-            }).map((item) => (
-              <Posted key={item.id} data={item} />
-            ))}
-          </div>
-        </div>
+      {filteredItems.posted && filteredItems.posted.length ? (
+        <ProductSection
+          headerText={t('my-list:card.posted')}
+          items={filteredItems.posted}
+          Card={Posted}
+        />
       ) : null}
 
-      {typedItems({
-        items: filteredItems,
-        byStatus: ProductStatus.DRAFT,
-      }).length ? (
-        <div css={styles.section}>
-          <div css={styles.header}>
-            <SectionHeader>{t('my-list:card.drafted')}</SectionHeader>
-          </div>
-          <div css={styles.container}>
-            {typedItems({
-              items: filteredItems,
-              byStatus: ProductStatus.DRAFT,
-            }).map((item) => (
-              <Drafted key={item.id} data={item} />
-            ))}
-          </div>
-        </div>
+      {filteredItems.drafts && filteredItems.drafts.length ? (
```

Comment:
`filteredItems.drafts.length ?` this check is enough

Reply:
nop